pkg/apis/core/v1/storagecluster.go

```diff
@@ -107,6 +107,8 @@ type StorageClusterSpec struct {
 	// Nodes node level configurations that will override the ones at cluster
 	// level. These configurations can be grouped based on label selectors.
 	Nodes []NodeSpec `json:"nodes,omitempty"`
+	// Resource requirements for portworx container in a storage cluster pod, e.g. CPU and memory requests or limits
+	Resources v1.ResourceRequirements `json:"resources,omitempty"`
```

Comment:
Let's make this a pointer `*v1.ResourceRequirements`. That way if it is empty, it does not show up at all in the yaml instead of an empty object - `resources: {}`.